app/jobs/finalize_import_job.rb

```diff
@@ -0,0 +1,37 @@
+class FinalizeImportJob < ActiveJob::Base
+  attr_reader :import
+
+  def perform(import_id)
+    @import = Import.find_by id: import_id
+    return unless import
+
+    broadcast_result
+
+    export_errors
+    post_to_slack
+
+    import.finish
+    broadcast_result
+  end
+
+  private
+
+  def import_result
+    @import_result ||= ImportResult.new(@import)
+  end
+
+  def broadcast_result
+    channel = "import_#{@import.id}"
+    ActionCable.server.broadcast channel, import_result
+  end
+
+  def export_errors
+    ImportErrors.export(@import)
+  end
+
+  def post_to_slack
+    url = Rails.application.routes.url_helpers.import_url(@import)
```

Comment:
I don't think this is bad, but at the same time I'm a bit concerned about globally changing the `host` in `config/environments/**.rb`. One thing you could do is to explicitly specify the `host`:

```ruby
Rails.application.routes.url_helpers.import_url(@import, host: 'http://www.bestdomainever.com')
```

If this is a common pattern in Job classes, we could have the parent `ApplicationJob` class include the entire `Rails.application.routes.url_helpers` module and set a `host`:

```ruby
class ApplicationJob < ActiveJob::Base
  include Rails.application.routes.url_helpers
  default_url_options[:host] = 'http://www.bestdomainever.com' # or use ENV['DEFAULT_HOST']
end
```

So in the `post_to_slack` method you could simply do:

```ruby
SlackBot.post(":tada: #{import_result.name} finished: #{import_url(@import)}")
```

Either way, `ENV['DEFAULT_HOST']` still needs to be set.